tahoe_sites/api.py

```diff
@@ -116,3 +122,114 @@ def is_active_admin_on_organization(user, organization):
         is_active=True,
         is_admin=True,
     ).exists()
+
+
+def create_tahoe_site_by_link(organization, site):
+    """
+    Link a site to an organization. Together they are a tahoe-site
+
+    :param organization: Organization object to be linked
+    :param site: Site object to be linked
+    :return: TahoeSite object
+    """
+    if zd_helpers.should_site_use_org_models():
+        organization.sites.add(site)
+        return None
+    return TahoeSite.objects.create(organization=organization, site=site)
+
+
+def create_tahoe_site(domain, short_name, uuid=None):
+    """
+    Centralized method to create the site objects in both `tahoe-sites` and `edx-organizations`.
+    Other pieces like SiteConfigurations are out of the scope of this helper.
+
+    :param domain: Site domain.
+    :param short_name: Organization short name, course key component and site name.
+    :param uuid: UUID string or object. Used to identify organizations and sites across Tahoe services.
+    :return: dict with `site` `organization` and `uuid` fields.
+    """
+    organization_data = {
+        'name': short_name,
+        'short_name': short_name,
+        'description': 'Organization of {domain} (automatic)'.format(domain=domain),
+    }
+
+    if uuid and zd_helpers.should_site_use_org_models():
+        organization_data['edx_uuid'] = uuid
+
+    with transaction.atomic():
```

Comment:
Internal APIs shouldn't use `transaction.atomic()`. Django commands and Django views should.

That's the assumption all Open edX have, with some few exceptions.

Even if a function uses `transaction.atomic()`, it have an alternative function without a transaction.

In the scope of this package, I think this shouldn't be used.

```suggestion
```

Reply:
Done